src/db/pool_prepared_statement.cr

```diff
@@ -39,16 +42,22 @@ module DB
         conn.release
         raise ex
       end
-      @connections << WeakRef.new(conn) unless existing
+      unless existing
```

Comment:
Can line 38 above be problematic? we pass unguarded the set of connections to `checkout_some`, but it can change mid-way, can't it?